jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -479,10 +479,10 @@ private static class RetainedBucket
         private RetainedBucket(int capacity, int poolSize)
         {
             if (poolSize <= ConcurrentPool.OPTIMAL_MAX_SIZE)
-                _pool = new ConcurrentPool<>(ConcurrentPool.StrategyType.THREAD_ID, poolSize, true);
+                _pool = new ConcurrentPool<>(ConcurrentPool.StrategyType.THREAD_ID, poolSize, false);
```

Comment:
@lorban I like this change as THREAD_ID and the ThreadLocal cache are kind of trying to do the same thing anyway.
However, I wonder if we could improve the THREAD_ID algorithm a little bit.  Currently it is used as:
```
            case THREAD_ID -> (int)(Thread.currentThread().getId() % size);
```

The problem here is that typically the threads in use will be consecutive IDs, so if there are 50 threads in use and the pool is 100 in size, it may be that 50 consecutive entries are all in use and that an acquire by a thread that already has used one buffer will now need to scan over 49 others to find an empty slot.  Perhaps we should be treating it a little more like an initial hash and spreading the thread ID over the available size more.    Perhaps the following would work a little bit better:

```
            case THREAD_ID -> (int)((Thread.currentThread().getId() * 31) % size);
```

(31 selected because it is prime)

Reply:
I modified the THREAD_ID strategy to multiply by 31, ran the benchmark, and measured no noticeable change.

Using the thread id as an index combined with the randomness of the OS' scheduler is already spreading the threads sufficiently well, it seems.

Reply:
Still it might be best to keep the * 31.   It is probably rare that all threads need buffers at exactly the same time, but I also think it could occur.  It may be the benchmark is not the right pattern to make it happen.   So you saw no cost to doing this, so let's keep it.

Reply:
I wrote a micro benchmark to see exactly what's going on and the * 31 costs exactly two cheap instructions: shift left by 5, then subtract once.

No big deal, so I'm okay with this change.